corehq/apps/locations/static/locations/js/location_drilldown.async.js

```diff
@@ -9,19 +9,20 @@ function api_get_children(loc_uuid, callback, loc_url) {
     });
 }
 
-function LocationSelectViewModel(hierarchy, default_caption, auto_drill, loc_filter, func, show_location_filter, loc_url) {
+function LocationSelectViewModel(options) {
     var model = this;
 
-    this.default_caption = default_caption || 'All';
-    this.auto_drill = (_.isBoolean(auto_drill) ? auto_drill : true);
-    this.loc_filter = loc_filter || function(loc) { return true; };
-    this.func = typeof func !== 'undefined' ? func : LocationModel;
-    this.show_location_filter = ko.observable((typeof show_location_filter !== 'undefined') ? show_location_filter : 'y');
+    this.loc_url = options.loc_url;
+    this.default_caption = options.default_caption || 'All';
+    this.auto_drill = (_.isBoolean(options.auto_drill) ? options.auto_drill : true);
+    this.loc_filter = options.loc_filter || function(loc) { return true; };
```

Comment:
@sravfeyn are you sure you need that `loc` argument? i think this might work without the argument as well or is that good to have for some browsers?